integration-tests/contracts/ccipreader_test.go

```diff
@@ -179,8 +191,116 @@ func TestCCIPReader_CommitReportsGTETimestamp(t *testing.T) {
 	assert.Equal(t, uint64(90), reports[0].Report.PriceUpdates.GasPriceUpdates[0].GasPrice.Uint64())
 }
 
+func TestCCIPReader_CommitReportsGTETimestamp_RespectsFinality(t *testing.T) {
+	t.Parallel()
+	ctx := tests.Context(t)
+	var finalityDepth int64 = 10
+	s, _, onRampAddress := setupGetCommitGTETimestampTest(ctx, t, finalityDepth)
+
+	tokenA := common.HexToAddress("123")
+	const numReports = 5
+
+	firstReportTs := emitCommitReports(ctx, t, s, numReports, tokenA, onRampAddress)
+
+	// Need to replay as sometimes the logs are not picked up by the log poller (?)
+	// Maybe another situation where chain reader doesn't register filters as expected.
+	require.NoError(t, s.lp.Replay(ctx, 1))
+
+	var reports []plugintypes.CommitPluginReportWithMeta
+	var err error
+	// Will not return any reports as the finality depth is not reached.
+	require.Never(t, func() bool {
+		reports, err = s.reader.CommitReportsGTETimestamp(
+			ctx,
+			chainD,
+			// Skips first report
+			//nolint:gosec // this won't overflow
+			time.Unix(int64(firstReportTs)+1, 0),
+			10,
+		)
+		require.NoError(t, err)
+		return len(reports) == numReports-1
+	}, tests.WaitTimeout(t)/2, 50*time.Millisecond)
+
+	// Commit finality depth number of blocks.
+	for i := 0; i < int(finalityDepth); i++ {
+		s.sb.Commit()
+	}
+
+	require.Eventually(t, func() bool {
+		reports, err = s.reader.CommitReportsGTETimestamp(
+			ctx,
+			chainD,
+			// Skips first report
+			//nolint:gosec // this won't overflow
+			time.Unix(int64(firstReportTs)+1, 0),
+			10,
+		)
+		require.NoError(t, err)
+		return len(reports) == numReports-1
+	}, tests.WaitTimeout(t)/2, 50*time.Millisecond)
+
+	assert.Len(t, reports, numReports-1)
+	assert.Len(t, reports[0].Report.MerkleRoots, 1)
+	assert.Equal(t, chainS1, reports[0].Report.MerkleRoots[0].ChainSel)
+	assert.Equal(t, onRampAddress.Bytes(), []byte(reports[0].Report.MerkleRoots[0].OnRampAddress))
+	assert.Equal(t, cciptypes.SeqNum(10), reports[0].Report.MerkleRoots[0].SeqNumsRange.Start())
+	assert.Equal(t, cciptypes.SeqNum(20), reports[0].Report.MerkleRoots[0].SeqNumsRange.End())
+	assert.Equal(t, "0x0200000000000000000000000000000000000000000000000000000000000000",
+		reports[0].Report.MerkleRoots[0].MerkleRoot.String())
+	assert.Equal(t, tokenA.String(), string(reports[0].Report.PriceUpdates.TokenPriceUpdates[0].TokenID))
+	assert.Equal(t, uint64(1000), reports[0].Report.PriceUpdates.TokenPriceUpdates[0].Price.Uint64())
+	assert.Equal(t, chainD, reports[0].Report.PriceUpdates.GasPriceUpdates[0].ChainSel)
+	assert.Equal(t, uint64(90), reports[0].Report.PriceUpdates.GasPriceUpdates[0].GasPrice.Uint64())
+}
+
+//func TestCCIPReader_GetLatestPriceSeqNr(t *testing.T) {
```

Comment:
Should it be uncommented

Reply:
It's intentional because cl-ccip is not merged yet.

Reply:
You can uncomment and skip the test with the ticket detail that can unskip it.

Reply:
removed completely for now